hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java

```diff
@@ -2598,6 +2598,9 @@ protected void consumeFromClauseCorrelatedRoot(SqmRoot<?> sqmRoot) {
 			final TableGroup parentTableGroup = fromClauseIndex.findTableGroup(
 					from.getCorrelationParent().getNavigablePath()
 			);
+			if ( parentTableGroup == null ) {
+				throw new InterpretationException( "Access to from node '" + from.getCorrelationParent() + "' is not possible in from-clause subqueries, unless the 'lateral' keyword is used for the subquery!" );
+			}
```

Comment:
I used to put exclamation marks in error messages back in the early days of Hibernate, but I don't think we need them anymore. 😆